src/parse.rs

```diff
@@ -621,6 +642,20 @@ fn parse_decimal(s: &str, ix: usize) -> Option<(usize, usize)> {
     usize::from_str(&s[ix..end]).ok().map(|val| (end, val))
 }
 
+// finds the an ID between < and > and returns (id, skip) where skip
+// is how far ahead we should jump ix (how much of the string we used)
+fn parse_id(s: &str, ix: usize) -> Option<(&str, usize)> {
+    if let Some(start) = s[ix..].find('<') {
+        if let Some(end) = s[ix..].find('>') {
```

Comment:
Does Oniguruma have any restrictions on what is allowed as the name?

Reply:
Yes, I believe group names need to start with a letter and then can only be letters and numbers. I'll find out for sure and tighten this up.

Reply:
Group names have to be made of \w's and/or underscores. I tried to make things work correctly with unicode by using Rust's char_indices function to iterate and is_alphanumeric() to check.